linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/IstioLogger.scala

```diff
@@ -0,0 +1,129 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.buoyant.H2
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.finagle._
+import com.twitter.io.Buf
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.config.types.Port
+import io.buoyant.k8s.istio.MixerClient
+import io.buoyant.linkerd.LoggerInitializer
+import io.buoyant.linkerd.protocol.HttpLoggerConfig
+import io.buoyant.router.context.DstBoundCtx
+
+object IstioLogger {
+  val unknown = "unknown"
+
+  // expected DstBoundCtx.current:
+  // Some((Path(%,io.l5d.k8s.daemonset,default,incoming,l5d,#,io.l5d.k8s.istio,reviews.default.svc.cluster.local,az:us-west::env:prod::version:v1,http),Path()))
+  val pathLength = 10
+  val pathServiceIndex = 7
+  val pathLabelsIndex = 8
+}
+
+class IstioLogger(mixerClient: MixerClient, params: Stack.Params) extends Filter[Request, Response, Request, Response] {
+  import IstioLogger._
+
+  private[this] val log = Logger.get()
+
+  def apply(req: Request, svc: Service[Request, Response]) = {
+    val start = Time.now
+
+    svc(req).respond { ret =>
+      val duration = Time.now - start
```

Comment:
code in finagle for this kind of thing usually uses `com.twitter.util.Stopwatch`.  Not sure what the difference is.... maybe stopwatch is more accurate?